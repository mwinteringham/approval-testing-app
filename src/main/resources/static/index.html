<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Approval testing practise app</title>
    <style>
        img {
            width: 600px;
            height: 310px;
        }

        .container {
            width: 600px;
            margin-left: auto;
            margin-right: auto;
            margin-top: 75px;
            border: 1px solid black;
            padding: 10px;
        }

        .details {
            color: navy;
            text-align: right;
            float: right;
        }

        .buttonHolder {
            text-align: center;
        }

        button {
            font-size: 20px;
        }
    </style>
    <script src="./jquery.js"></script>
    <script>
        $.get( "/car", function( data ) {
            $("#image").attr("src", data.image);
            $("#company").html(data.company);
            $("#make").html(data.make);
            $("#price").html(data.price);
            $("#purchase").html(data.datePurchased);
        });

        var postToggle = function(){
            $.post( "/toggle", function( data ) {
                location.reload();
            });
        }
    </script>
</head>
<body>
    <div class="container">
        <img src="" id="image" />
        <h1>Company: <div class="details" id="company">Unknown</div></h1>
        <h1>Make: <div class="details" id="make">Unknown</div></h1>
        <h1>Price: <div class="details" id="price">Unknown</div></h1>
        <h1>Purchase date: <div class="details" id="purchase">Unknown</div></h1>
        <div class="buttonHolder">
            <button onclick="postToggle()">Toggle</button>
        </div>
    </div>
</body>
</html>