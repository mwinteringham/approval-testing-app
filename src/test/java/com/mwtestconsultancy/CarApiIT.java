package com.mwtestconsultancy;

import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.springframework.boot.SpringApplication;
import org.springframework.context.ApplicationContext;

import static io.restassured.RestAssured.given;

public class CarApiIT {

    private ApplicationContext app;

    @Before
    public void setupApi(){
        app = SpringApplication.run(CarApi.class);
    }

    @After
    public void killApi(){
        SpringApplication.exit(app);
    }

    @Test
    public void testGetCar(){
        Response carResponse = given()
                                .get("http://localhost:8080/car");

        Approvals.verify(carResponse.getBody().print());
    }

    @Test
    public void testToggleCar(){
        Response toggleResponse = given()
                                    .post("http://localhost:8080/toggle");

        Approvals.verify(toggleResponse.getStatusCode());
    }

    @Test
    public void testToggleCarSendsNewPayload(){
        Response toggleResponse = given()
                .post("http://localhost:8080/toggle");

        Response carResponse = given()
                .get("http://localhost:8080/car");

        Approvals.verify(carResponse.getBody().print());
    }

}
