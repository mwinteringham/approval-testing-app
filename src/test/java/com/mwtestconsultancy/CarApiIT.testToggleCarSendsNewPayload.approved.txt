{"company":"Toyota","make":"Auris","price":28000,"datePurchased":"2017-12-15","image":"https://upload.wikimedia.org/wikipedia/commons/thumb/5/5a/2016_Toyota_Corolla_%28ZRE182R%29_Ascent_Sport_hatchback_%282017-01-30%29_01.jpg/1600px-2016_Toyota_Corolla_%28ZRE182R%29_Ascent_Sport_hatchback_%282017-01-30%29_01.jpg"}
